doc/users/style_sheets.rst

```diff
@@ -25,15 +25,15 @@ Defining your own style
 =======================
 
 You can create custom styles and use them by calling ``style.use`` with the
-path or URL to the style sheet. Alternatively, if you add your
-``<style-name>.mplstyle`` file to ``~/.matplotlib/stylelib`` (you may need to
-create this directory), you can reuse your custom style sheet with a call to
-``style.use(<style-name>)``. Note that a custom style sheet in
-``~/.matplotlib/stylelib`` will override a style sheet defined by matplotlib if
-the styles have the same name.
+path or URL to the style sheet. Alternatively, if you add your ``<style-name>.mplstyle`` 
+file to ``mpl_configdir/stylelib`, you can reuse your custom style sheet with a call to 
+``style.use(<style-name>)``. By default ``mpl_configdir`` should be ``~/.config/matplotlib``, 
```

Comment:
@tacaswell @tonysyu 
I had checked that on Linux at home.

Now, at work and at a windows7 box I get `/.matplotlib` so @tonysyu is right. It seems the only way is to use `matplotlib.get_configdir()`.

Thanks.